File: backend/feed-service/src/main/java/ru/maxigram/backend/feedservice/service/PostService.java
package ru.maxigram.backend.feedservice.service;

import java.util.Collection;
import java.util.Optional;
import ru.maxigram.backend.feedservice.entity.Post;

/** Сервис для работы с постами. */
public interface PostService {

  /**
   * Создаёт пост.
   *
   * @param authorId идентификатор автора
   * @param text текст
   * @return созданный пост
   */
  Post createPost(String authorId, String text);

  /**
   * Ищет пост по идентификатору.
   *
   * @param id идентификатор поста
   * @return {@link Optional} с постом с указанным идентификатором
   */
  Optional<Post> findPostById(Long id);

  /**
   * Ищет посты заданных авторов.
   *
   * @param authorIds идентификаторы авторов
   * @return посты заданных авторов
   */
  Collection<Post> findPostsByAuthorIds(Collection<String> authorIds);
}

File: src/main/kotlin/org/maxizenit/maxigram/feed/PostService.kt
package org.maxizenit.maxigram.feed

import org.springframework.stereotype.Service
import java.time.Clock
import java.time.Instant
import java.util.UUID

@Service
class PostService(
    private val posts: PostRepository,
    private val clock: Clock = Clock.systemUTC(),
) {

    fun create(authorId: UUID, text: String): Post {
        val trimmed = text.trim()
        if (trimmed.isEmpty()) throw InvalidContentException("Post text must not be blank")
        return posts.insert(authorId, trimmed, Instant.now(clock))
    }

    fun getView(id: Long, requesterId: UUID): PostView =
        posts.viewById(id, requesterId) ?: throw PostNotFoundException(id)
}
